os/board/rtl8720e/src/component/wifi/api/wifi_conf_ext_inic_ipc.c

```diff
@@ -883,4 +887,53 @@ unsigned int wifi_get_tsf_low(unsigned char port_id)
 	return inic_ipc_host_get_wifi_tsf_low(port_id);
 }
 
+//----------------------------------------------------------------------------//
+int wifi_csi_config(rtw_csi_action_parm_t *act_param)
+{
+	int ret = 0;
+	u32 param_buf[1];
+
+	param_buf[0] = (u32)act_param;
+	DCache_Clean((u32)act_param, sizeof(rtw_csi_action_parm_t));
+	ret = inic_ipc_api_host_message_send(IPC_API_WIFI_CONFIG_CSI, param_buf, 1);
+	return ret;
+}
+
+int wifi_csi_report(u32 buf_len, u8 *csi_buf, u32 *len, rtw_csi_header_t *csi_header)
+{
+	int ret = 0;
+	u32 param_buf[4];
+
+	void *csi_buf_temp = rtw_zmalloc(buf_len);
+	rtw_csi_header_t *csi_header_temp = (rtw_csi_header_t *)rtw_malloc(sizeof(rtw_csi_header_t));
+	u32 *len_temp = (u32 *)rtw_zmalloc(sizeof(u32));
+	if (csi_buf_temp == NULL || csi_header_temp == NULL || len_temp == NULL) {
+		return -1;
```

Comment:
Is it okay not to free that successfully allocated?
I mean,
if success to allocate `csi_buf_temp`, but fail to allocate `csi_header_temp`, it is better to free the `csi_buf_temp` before returning `-1` in line 911.

Reply:
you're right, let me do some modifications on this